src/common/timer_manager.cc

```diff
@@ -0,0 +1,63 @@
+#include "timer_manager.h"
+
+#include "common/util.h"
+
+namespace raftcpp {
+namespace common {
+
+TimerManager::TimerManager() {
+    io_service_ = std::make_unique<asio::io_service>();
+    work_ = std::make_unique<asio::io_service::work>(*io_service_);
+}
+
+TimerManager::~TimerManager() {
+    for (auto &timer : timers_) {
+        timer->Stop();
+    }
+
+    io_service_->stop();
+    thread_->join();
+}
+
+void TimerManager::Run() {
+    // Note that the `Start()` should be invoked before `io_service->run()`.
+    thread_ = std::make_unique<std::thread>([this]() { io_service_->run(); });
+}
+
+void TimerManager::StartTimer(int id, uint64_t timeout_ms) {
+    int size = (int)timers_.size();
+    if (id >= 0 && id < size) {
```

Comment:
I think id shouldn't be out of the range(0-size), so is it make more sense that:
RAFTCPP_CHECK(0 <= id <= size);

Reply:
OK